### ISSUE TITLE
HttpRequest with leading \r before request-line rejected with BadRequestException

### ISSUE THREAD

**Issue by suwatch** [dotnet/aspnetcore#40747]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

HttpRequest with leading \r before request-line rejected with BadRequestException

### Expected Behavior

Should support (simply skip) similar to IIS or other server implementation.

https://stackoverflow.com/questions/46924558/http-leading-blank-line-in-header-a-valid-http-request-header

### Steps To Reproduce

```
00000000  0d 50 4f 53 54 20 2f 50  6f 73 74 2f  .POST /Post/
```

By using socket to simulate the leading \r before request-line.

```c#
var url = new Uri(args[0]);
int hostPort = 80;
var entry = await Dns.GetHostEntryAsync(url.Host);
var hostIp = entry.AddressList.FirstOrDefault();
var hostep = new IPEndPoint(hostIp, hostPort);
using (var sock = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
{
    await sock.ConnectAsync(hostep);
    var strb = new StringBuilder();
    // pass leading \r
    strb.Append($"\rGET / HTTP/1.1\r\n");
    strb.Append($"HOST: {url.Host}\r\n");
    strb.Append($"Connection: close\r\n");
    strb.Append($"User-Agent: testclient/1.1\r\n");
    strb.Append("\r\n");

    Console.WriteLine(strb);
    sock.Send(Encoding.UTF8.GetBytes(strb.ToString()));

    var buf = new byte[4096];
    var bytes = sock.Receive(buf, buf.Length, 0);
    Console.WriteLine(Encoding.ASCII.GetString(buf, 0, bytes));
    sock.Close();
}
```

### Exceptions (if any)

```c#
krpbfe: Received a bad request from client IP ::ffff:205.154.150.223 port 9073 for sni , error: Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException: Invalid request line: ''
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpParser`1.GetUnknownMethodLength(ReadOnlySpan`1 span)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpParser`1.ParseRequestLine(TRequestHandler handler, ReadOnlySpan`1 requestLine)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpParser`1.ParseRequestLine(TRequestHandler handler, SequenceReader`1& reader)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1Connection.TakeStartLine(SequenceReader`1& reader)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1Connection.ParseRequest(SequenceReader`1& reader)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1Connection.TryParseRequest(ReadResult result, Boolean& endConnection)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequestsAsync[TContext](IHttpApplication`1 application)
```

### .NET Version

6.0.102

**Comment by adityamandaleeka** [comment]
Reopening for servicing

**Comment by adityamandaleeka** [comment]
This was fixed in 6.0.5. Closing.
